compiler/qsc_circuit/src/circuit.rs

```diff
@@ -96,7 +96,7 @@ impl Config {
 }
 
 type ObjectsByColumn = FxHashMap<usize, CircuitObject>;
-type ColumnWidthsByColumn = FxHashMap<usize, usize>;
+type ColumnWidthsByColumn = FxHashMap<usize, ColumnRenderer>;
```

Comment:
I like how you organized this. I just think we can simplify the names.

```suggestion
type FullColumns = FxHashMap<usize, Column>;
```

That's what the new type is... it's just the column! No need to overthink it imo.

And now we can think of the hashmap as containing only the columns that have objects in them, so I suggested a name that reflects that.

Reply:
I'm not 100% sure I agree with this:

My thinking is that the "ByColumn" postfix is consistent with `ObjectsByColumn` in the row above, because both are a hashmap with the key being the column index. 

And the `ColumnRenderer` itself is a struct that takes circuit objects and turns them into ASCII art, so calling it a `Column` sounds a bit confusing to me. 

Not to mention how, if you agree with my point about `ByColumn` but don't agree with this point, then we'll call this hashmap `ColumnByColumn`, and that's just silly. :)

Having said that, I will of course rename both of these if you still think your suggested names are better.

Reply:
Actually, I realize now that I can resolve most of this comment and the other open one pretty easily, it's an oversight on my side - this alias isn't necessary, because we don't actually need a hashmap for this. Let me try it out... 

You'll still have to decide between `Column` and `ColumnRenderer`, though.

Reply:
Okay, short answer: Totally agree with you on the bit where you say it would be inconsistent with `ObjectsByColumn` map right above this - I hadn't noticed that. Consistency trumps all, so let's stick with your version. _Maybe_ consider `ColumnWidth` instead of `ColumnRenderer`.

---

Long answer: humor me for a second while I try to explain what I originally had in mind. Naming discussions tend to get pretty philosophical.

- "column index": it's just a number and that's what the original hashmap index `usize` refers to.
- "column": that's the struct that describes the "physical" column in the diagram, so to speak. It contains attributes of the column, such as its width. 

Personally I'm biased against struct names like "Something-Doer" because I prefer structs to have names that describe what the thing is, not what it does.

I guess then the natural conclusion would have been to rename these to `ObjectsByColumnIndex` and `ColumnsByColumnIndex` ... and then rename all the other variables that refer to `column`s below to `column_index` ... and I mean... that _is_ getting into silly territory.

Reply:
Yeah, that works - please look at the latest commit. Basically, since the column renderers are made by iterating over `0..end_column`, they can just be put in a vector. That means there's no need for this hashmap at all.

I can still rename the `ColumnRenderer` to `Column`, though then the enumeration in line 192 and 200 will be over `(column_index, column)` instead of the current `(column, renderer)`. Not sure what's clearer in this case.

Reply:
I see I managed to miss the short-answer-long-answer comment you made a couple of days ago (that is - when last I loaded this page, I had 3 comments in a row, which made for a slightly more coherent conversation with myself. 😄).

You convinced me that `ColumnRenderer` isn't ideal. `ColumnWidth` bothers me a bit, though, because I find code like `column_width.fmt_on_..._wire` to be confusing. So, given that there is already a `Row` struct in this file, I tried `Column` to see what happens.

I find that it makes the meaning of `column` a bit ambiguous - sometimes it's this struct, and sometimes it's still the index (e.g. in `Wire::Classical { start_column }`). However, it's very clear from the context what meaning is used when, and mistakes are easily caught by the compiler because of the types. So I can definitely live with it, and I renamed it in the latest commit.